controllers/linklist/CategoryController.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+namespace humhub\modules\smartVillage\controllers\linklist;
+
+use humhub\components\access\ControllerAccess;
+use humhub\modules\admin\permissions\ManageSpaces;
+use humhub\modules\content\components\ContentContainerController;
+use humhub\modules\content\components\ContentContainerControllerAccess;
+use humhub\modules\linklist\models\Category;
+use humhub\modules\smartVillage\components\AuthBaseController;
+use humhub\modules\smartVillage\components\BaseContentContainerController;
+use humhub\modules\space\models\Space;
+use humhub\modules\user\models\User;
+use yii\web\HttpException;
+use Yii;
+
+class CategoryController extends BaseContentContainerController
+{
+
+    /** access level of the user currently logged in. 0 -> no write access / 1 -> create links and edit own links / 2 -> full write access. * */
+    public $accessLevel = 0;
+
+    public function getAccessRules()
+    {
+        return [
+            [ContentContainerControllerAccess::RULE_USER_GROUP_ONLY => [Space::USERGROUP_MEMBER, User::USERGROUP_SELF]],
+            [ControllerAccess::RULE_PERMISSION => [ManageSpaces::class], 'actions' => ['config']],
+        ];
+    }
+
+    /**
+     * Automatically loads the underlying contentContainer (User/Space) by using
+     * the uguid/sguid request parameter
+     */
+    public function init()
+    {
+        parent::init();
+        $this->accessLevel = $this->getAccessLevel();
+    }
+
+    /**
+     * Get the acces level to the linklist of the currently logged in user.
+     * @return number 0 -> no write access / 1 -> create links and edit own links / 2 -> full write access
+     */
+    private function getAccessLevel()
+    {
+        if ($this->contentContainer instanceof \humhub\modules\user\models\User) {
+            return $this->contentContainer->id == Yii::$app->user->id ? 2 : 0;
+        } else if ($this->contentContainer instanceof \humhub\modules\space\models\Space) {
+            return $this->contentContainer->can(new \humhub\modules\post\permissions\CreatePost()) ? 2 : 1;
+        }
+    }
+
+    public function actionIndex(){
+        $results = [];
+        $categoryBuffer = Category::find()->contentContainer($this->contentContainer)->orderBy(['sort_order' => SORT_ASC]);
+
+        if(empty($categoryBuffer->all())){
+            return $this->returnError(400, "Category not found");
+        }
+
+        $pagination = $this->handlePagination($categoryBuffer);
+
+
+        $space = Space::findOne(['guid' => Yii::$app->request->get('cguid')]);
+        $user = User::findOne(['guid' => Yii::$app->request->get('cguid')]);
```

Comment:
> Note : We have to pass guid of space or user as cguid param in every endpoint to the access the APIs. because in web guid of space or user is getting using get request.
> 
> ![image](https://user-images.githubusercontent.com/103625795/171170604-08d566bb-59ba-4fd9-a7da-36f41eab8e05.png)
> 
> You can suggest us which data will be return. As of now we are returning data as our understanding
> 

what do you mean exactly? i think that requests just work on the web because of some logged in cookies or something? if we call our APIs we would need to append the space or user guid on our own i guess from an earlier space or user request?

Reply:
Hi @donni106 
We have to append cguid of user or space in our APIs because it will check the user has permission to access this space or user.

Reply:
i understand. but what is `Yii::$app->request->get('cguid')` doing. how does it work. what `cguid` is fetched from where?

Reply:
Hi @donni106 
Yii::$app->request->get('cguid') is used to fetch value of guid of space or user that we are passing in query param. it is similar to $_GET we are used in PHP.

![image](https://user-images.githubusercontent.com/103625795/171791323-4608f0ff-2584-471a-aaf1-ed088fdca6bc.png)